sdk/python/tests/compiler/testdata/artifact_location.yaml

```diff
@@ -45,37 +49,40 @@ spec:
       valueFrom:
         secretKeyRef:
           key: accesskey
-          name: minio
+          name: $(inputs.params.secret_name)
```

Comment:
Hi Tommy, 

this seems to not work out-of-the box. Pipeline does not show any progress, there are no logs and nothing stored in Minio.

`$ tkn pipeline start custom-artifact-location-pipeline --showlog`:
```
? Value for param `secret_name` of type `string`? (Default is `mlpipeline-minio-artifact`) mlpipeline-minio-artifact
? Value for param `tag` of type `string`? (Default is `1.31.0`) 1.31.0
? Value for param `namespace` of type `string`? (Default is `kubeflow`) kubeflow
? Value for param `bucket` of type `string`? (Default is `mlpipeline`) mlpipeline
Pipelinerun started: custom-artifact-location-pipeline-run-6mcxt
Waiting for logs to be available...
^C
```

... and later ...
`$ tkn pipeline logs custom-artifact-location-pipeline --last`:
```
Pipeline still running ...
No logs found
```
---
But when I change it back to 
```
    - name: AWS_ACCESS_KEY_ID
      value: minio
    - name: AWS_SECRET_ACCESS_KEY
      value: minio123
```

Then it works and I get this output:
```
? Value for param `secret_name` of type `string`? (Default is `mlpipeline-minio-artifact`) mlpipeline-minio-artifact
? Value for param `tag` of type `string`? (Default is `1.31.0`) 1.31.0
? Value for param `namespace` of type `string`? (Default is `kubeflow`) kubeflow
? Value for param `bucket` of type `string`? (Default is `mlpipeline`) mlpipeline
Pipelinerun started: custom-artifact-location-pipeline-run-d2x6l
Waiting for logs to be available...

[foo : copy-artifacts] Added `storage` successfully.
[foo : copy-artifacts] `/tekton/results/output` -> `storage/mlpipeline/runs/custom-artifact-location-pipeline-run-d2x6l/custom-artifact-location-pipeline-run-d2x6l-foo-2v5gf-pod-86ll6/output.txt`
[foo : copy-artifacts] Total: 0 B, Transferred: 6 B, Speed: 565 B/s
```

Reply:
For this pipeline, you have to run under the same namespace as kfp, so by default you should run it under the kubeflow namespace. If you want to run on different namespace, then you have to create a secret with the s3 key and secret using the same `secret_name` in the parameter.

Reply:
I am thinking from a general integration test suite, it would be great if we can run this (and all other) pipeline as is. In order to use Tekton I am using a different default namespace. Can we just use the `minio` and `minio123` credentials since they are not really secret?

Reply:
The DSL is designed to take the credentials from secret, even the default setting is reading from a secret. Is is possible for you to create/copy the `mlpipeline-minio-artifact` from kubeflow to your default namespace?

Reply:
For automation this would not be practical. Maybe we can run all pipelines in the `kubeflow` namespace and clean up afterwards.

Reply:
A lot of CI tests in kubeflow assume the kubernetes namespace is setup with all the necessary dependencies such as gcp secret, image pull secret, service account, etc. Probably we should create some sort of CI setup script for Tekton as well. Many of the design decision I made is to align with how Argo behaves, so kfp can reuse the same set up for Tekton.